src/providers/Web3Provider.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import { WagmiProvider, createConfig, http } from "wagmi";
+import {
+  arbitrum,
+  base,
+  mainnet,
+  optimism,
+  polygon,
+  sepolia,
+} from "wagmi/chains";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ConnectKitProvider, getDefaultConfig } from "connectkit";
+import type { ReactNode } from "react";
+
+const config = createConfig(
+  getDefaultConfig({
+    // Your dApps chains
+    chains: [mainnet, optimism, arbitrum, base, polygon, sepolia],
+    transports: {
```

Comment:
if we really want to set up RPCs, `#would` wrap this with `fallback` like mentioned in the docs: https://docs.family.co/connectkit/rpc-providers#rpc-providers